### ISSUE TITLE
clone() silently drops **algorithm_params, so tuned estimators revert to defaults inside GridSearchCV

### ISSUE THREAD

**Issue by ChrisGVE** [ChrisGVE/localdata-mcp#30]
## Summary

The `AnalysisPipelineBase` transformers accept algorithm-specific keyword arguments through `**algorithm_params` and store them on `self.algorithm_params`. `BaseEstimator.get_params()` introspects the `__init__` signature, so those parameters are never reported — and `clone()` therefore cannot restore them.

The result is a **silent** loss of configuration, not an error.

## Reproduction

```python
from sklearn.base import clone
from localdata_mcp.domains.pattern_recognition import ClusteringTransformer

c = ClusteringTransformer(algorithm="dbscan", eps=0.77)
print(c.algorithm_params)          # {'eps': 0.77}
print(clone(c).algorithm_params)   # {}          <-- eps is gone
print(sorted(c.get_params()))
# ['algorithm', 'auto_k_selection', 'k_range', 'n_clusters', 'random_state', 'standardize']
# no 'algorithm_params', no 'eps'
```

## Why this matters

`clone()` is not an exotic call. `GridSearchCV`, `RandomizedSearchCV`, `cross_val_score`, `cross_validate` and `Pipeline` used inside any of them clone every step before fitting. A DBSCAN configured with `eps=0.77` therefore runs with scikit-learn's default `eps` inside cross-validation, and reports a score for a model the caller never asked for. Nothing raises.

This violates the scikit-learn estimator contract, which requires that `get_params()` expose every constructor parameter and that `clone()` produce an equivalent unfitted estimator.

## Affected classes

Every `AnalysisPipelineBase` subclass taking `**algorithm_params` or equivalent. Confirmed on `ClusteringTransformer`; `DimensionalityReductionTransformer` and `AnomalyDetectionTransformer` follow the same pattern and should be checked.

## Suggested fix

Replace `**algorithm_params` with an explicit `algorithm_params: Optional[Dict[str, Any]] = None` constructor parameter, stored verbatim as `self.algorithm_params`. That makes it visible to `get_params()`, survivable through `clone()`, and tunable by name in a grid (`cluster__algorithm_params`). It also stops unknown keyword arguments being silently swallowed and forwarded into the wrapped estimator, which produces confusing errors such as `KMeans.__init__() got an unexpected keyword argument 'analytical_intention'`.

A regression test should assert `clone(est).get_params() == est.get_params()` for every transformer in the domain packages. `sklearn.utils.estimator_checks.check_estimator` would cover this and more.

## Context

Found while answering a question about how freely the pipeline layer composes. Note that this layer is **not** on the live MCP path — the MCP tools call the domain functions directly through the adapters — so no shipped tool is currently affected. It matters for anyone using the package as a Python library, and for any future tool that routes through the pipeline classes.

**Comment by ChrisGVE** [comment]
Closing as superseded by the v3 rebuild — and the distinction matters, so it is stated plainly rather than implied by the closure: **this was not fixed. The code it describes was deleted.**

`new-v3` is a ground-up rewrite. All 71 tools 2.1.0 registered are gone, replaced by eight verbs, and the package is now nine flat modules — no `domains/`, no `datascience_tools.py`, no `time_series_analysis/`. Nothing this issue points at exists in the tree any more, so there is no site left to repair and no reader who can act on it.

It stays closed unless 2.x is given a patch release, which is a decision nobody has taken. If that happens, reopen this rather than re-deriving it: the analysis here is still correct about the code it was written against.

Security exposures in the published 2.1.0 artifact are deliberately **not** being closed this way — #25, #33, #36 and #42 stay open, because PyPI still serves the artifact they describe and closing them would erase the only record of it.
